lib/rucio/richclient.py

```diff
@@ -0,0 +1,308 @@
+            if _force_cfg_log_level(cfg_opt):
+                logger_level = level
+
+    logger.setLevel(logger_level)
+
+    def add_handler(logger: logging.Logger,
+                    console: Optional[Console] = None,
+                    verbose: bool = False) -> None:
+
+        def time_formatter(timestamp: datetime) -> Text:
+            return Text(f"[{timestamp.isoformat(sep=' ', timespec='milliseconds')}]")
+
+        console = console or Console()
+        handler = RichHandler(rich_tracebacks=True, markup=True, show_path=verbose, show_time=verbose, console=console, tracebacks_width=min(console.width, MAX_TRACEBACK_WIDTH),
+                              tracebacks_word_wrap=True, log_time_format=time_formatter)
+        logger.addHandler(handler)
+
+    # Setting handler and formatter.
+    if not logger.handlers:
+        add_handler(logger, console, verbose)
+
+    return logger
+
+
+def generate_table(
+    rows: Sequence[Sequence[RenderableType]],
+    headers: Optional[Sequence[RenderableType]] = None,
+    row_styles: Optional[Sequence[StyleType]] = None,
+    col_alignments: Optional[list[JustifyMethod]] = None,
+    table_format: box.Box = CLITheme.TABLE_FMT,
+) -> Table:
+    """
+    Generates a Rich Table object from given input rows.
+
+    The elements in each row can be either plain strings or Rich renderable objects.
+    Passing strings will display them as simple text, while using Rich objects
+    allows you to introduce additional structure, styling, and widgets (e.g. Text, Trees) into
+    the table. Strings with style markup will be rendered as styled text.
+
+    :param table_format: style of the table
+    :param headers: list of headers
+    :param rows: list of rows
+    :param col_alignments: list of column alignments
+    :param row_styles: list of row styles
+    :returns: a Rich Table object
+    """
+    table = Table(box=table_format, show_header=headers is not None and len(headers) > 0)
+    table.row_styles = row_styles or ['none', 'dim']
+
+    if len(rows) == 0:
+        if headers:
+            for header in headers:
+                table.add_column(header)
+        return table
+
+    # Auto-detect on first row, numerical values on the right.
+    col_alignments = col_alignments or ['right' if str(col).isnumeric() else 'left' for col in rows[0]]
+    headers = headers or [''] * len(rows[0])
+
+    if col_alignments and len(headers) > len(col_alignments):
+        while len(headers) > len(col_alignments):
+            col_alignments.append('left')
+
+    headers = headers or [''] * len(rows[0])
+    for i, header in enumerate(headers):
+        table.add_column(header, overflow='fold', justify=col_alignments[i])
+
+    for row in rows:
+        row = ['' if str(col) == 'None' else col for col in row]
+        table.add_row(*row)
+    return table
+
+
+def print_output(
+    *output: Any,
+    console: Console,
+    no_pager: bool = False
+) -> None:
```

Comment:
The **Check Python Type Annotations** test currently fails because of the use of `Any` here. Not sure how to change it as `Console.print()` in `rich.console`  accepts `Any` as the type to the print statement too. 

Should I perhaps use `*output: object`  instead of `*output: Any`? That would lead to a passing test.

Reply:
No, I think you’re doing the right thing. The CI treats `Any` as missing a type, but there could be functions such as this where literally any type is permissible and handled properly.